### PR TITLE
Person Service update: Race Module

### PR DESCRIPTION
**Pull request by upasanapattnaik-eq**
Update to patient race model in Person Service:
+ Additional fields returned by stored procedure. 
+ Updated tests. 

Sprocs added to Liquibase module: 
+ Added sp_nrt_patient_postprocessing.
+ Added table creation scripts for nrt_patient and nrt_provider. 
+ Added sp_patient_race_event sproc.
+ Updated sp_patient_event sproc.

### CODE REVIEW

File: data-reporting-service/src/main/resources/db/sp-patient-001.sql
[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/src/main/resources/db/sp-patient-001.sql:1]: It returns a single row as a resultset for each value passed but yes, the expense of multiple joins, multiple calls within the select statement is valid.

[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/src/main/resources/db/sp-patient-001.sql:1]: The nrt_batch_log is currently setup for the post-processing sprocs which are batched. @rshanmugam-enquizit and I will be looking into error handling for the stored procedures/service later this sprint for the post-processing. 
For pre-processing, we are have the error message and id output by the java service. We could an additional table for pre-processing or modify the current nrt_batch_log to better identify logging.

File: data-reporting-service/src/main/resources/db/sp-patient-race-001.sql
[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/src/main/resources/db/sp-patient-race-001.sql:1]: Similar to answer above, logging can be determined based on aligned decision.

File: data-reporting-service/src/main/resources/db/sp-patient-race-001.sql [hunk 1]
```diff
@@ -0,0 +1,882 @@
...
+			 cte cte1
+where  sppr.PATIENT_UID = cte1.PATIENT_UID
+--and cte1.rn = 1
+;
+
+
+update #TMP_S_PERSON_ROOT_RACE
+set PATIENT_RACE_CALCULATED =
+        case
+            when len(PATIENT_RACE_CALC_DETAILS) <1 OR PATIENT_RACE_CALC_DETAILS is null then 'Unknown'
```
[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/src/main/resources/db/sp-patient-race-001.sql:156]: There is an ltrim on the variable before this statement. Will update with rtrim as well.

File: data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/PersonExtendedProps.java
[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/PersonExtendedProps.java:1]: I think multi-line comments are possible. Not sure about XML comments.

[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/model/dto/PersonExtendedProps.java:1]: For clarification, is this intended for each field or sub-catorgies eg. Race or Address?

File: data-reporting-service/src/main/resources/db/sp-patient-001.sql
[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/src/main/resources/db/sp-patient-001.sql:1]: Updated

File: data-reporting-service/src/main/resources/db/sp-patient-race-001.sql
[the remark this answers is not in the document]
Comment (upasanapattnaik-eq) [reply data-reporting-service/src/main/resources/db/sp-patient-race-001.sql:1]: Updated

**Review by rseshadri99 — APPROVED** [review]

File: data-reporting-service/src/main/resources/db/readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    * ```./gradlew update```
+
+### Liquibase Error Handling
+
+<b>ValidationFailedException</b>: When change is made to existing change log. This can be handled by removing the conflicting id or revert the changes.
+To remove the id, the following command can be run:
+
+```
+delete from database.dbo.databasechangelog where id = 'id'
```
Comment (alog-enquizit) [inline data-reporting-service/src/main/resources/db/readme.md:32]: 👍

**Review by alog-enquizit — APPROVED** [review]
lgtm 👍

**Pull request merged**